crypto/bn/asm/loongarch-mont.pl

```diff
@@ -0,0 +1,333 @@
+	slli.d  $num,$num,`log($BNSZ)/log(2)`
+	li.d	$t8,-4096
+    $SUBU   $sp,$sp,$num
+	and     $sp,$sp,$t8
+
+	$LD     $ahi,$ap,$BNSZ
+	$LD     $nhi,$np,$BNSZ
+	$MULD	$lo0,$aj,$bi
+	$MULHD	$hi0,$aj,$bi
+	$MULD	$m1,$lo0,$n0
+
+	$MULD	$alo,$ahi,$bi
+	$MULHD	$ahi,$ahi,$bi
+
+	$MULD	$lo1,$nj,$m1
+	$MULHD	$hi1,$nj,$m1
+	$ADDU	$lo1,$lo1,$lo0
+	sltu	$t8,$lo1,$lo0
+	$ADDU	$hi1,$hi1,$t8
+	$MULD	$nlo,$nhi,$m1
+	$MULHD	$nhi,$nhi,$m1
+
+	move	$tp,$sp
+	li.d	$j,2*$BNSZ
+.align	4
+.L1st:
+	$ADDU   $aj,$ap,$j
+	$ADDU   $nj,$np,$j
+	$LD     $aj,$aj,0
+	$LD     $nj,$nj,0
+
+	$ADDU	$lo0,$alo,$hi0
+	$ADDU	$lo1,$nlo,$hi1
+	sltu	$t8,$lo0,$hi0
+	sltu	$t0,$lo1,$hi1
+	$ADDU	$hi0,$ahi,$t8
+	$ADDU	$hi1,$nhi,$t0
+	$MULD	$alo,$aj,$bi
+	$MULHD	$ahi,$aj,$bi
+
+	$ADDU	$lo1,$lo1,$lo0
+	sltu	$t8,$lo1,$lo0
+	$ADDU	$hi1,$hi1,$t8
+	addi.d	$j,$j,$BNSZ
+	$ST     $lo1,$tp,0
+	sltu	$t0,$j,$num
+	$MULD	$nlo,$nj,$m1
+	$MULHD	$nhi,$nj,$m1
+
+	$PTR_ADD $tp,$tp,$BNSZ
+	bnez	$t0,.L1st
+
+	$ADDU	$lo0,$alo,$hi0
+	sltu	$t8,$lo0,$hi0
+	$ADDU	$hi0,$ahi,$t8
+
+	$ADDU	$lo1,$nlo,$hi1
+	sltu	$t0,$lo1,$hi1
+	$ADDU	$hi1,$nhi,$t0
+	$ADDU	$lo1,$lo1,$lo0
+	sltu	$t8,$lo1,$lo0
+	$ADDU	$hi1,$hi1,$t8
+
+	$ST     $lo1,$tp,0
+
+	$ADDU	$hi1,$hi1,$hi0
+	sltu	$t8,$hi1,$hi0
+	$ST     $hi1,$tp,$BNSZ
+	$ST     $t8,$tp,2*$BNSZ
+
+	li.d	$i,$BNSZ
+.align	4
+.Louter:
+	$ADDU   $bi,$bp,$i
+	$LD     $bi,$bi,0
+	$LD     $aj,$ap,0
+	$LD     $ahi,$ap,$BNSZ
+	$LD     $tj,$sp,0
+
+	$LD     $nj,$np,0
+	$LD     $nhi,$np,$BNSZ
+	$MULD	$lo0,$aj,$bi
+	$MULHD	$hi0,$aj,$bi
+	$ADDU	$lo0,$lo0,$tj
+	sltu	$t8,$lo0,$tj
+	$ADDU	$hi0,$hi0,$t8
+	$MULD	$m1,$lo0,$n0
+
+	$MULD	$alo,$ahi,$bi
+	$MULHD	$ahi,$ahi,$bi
+
+	$MULD	$lo1,$nj,$m1
+	$MULHD	$hi1,$nj,$m1
+
+	$ADDU	$lo1,$lo1,$lo0
+	sltu	$t8,$lo1,$lo0
+	$ADDU	$hi1,$hi1,$t8
+	$MULD	$nlo,$nhi,$m1
+	$MULHD	$nhi,$nhi,$m1
+
+	move	$tp,$sp
+	li.d	$j,2*$BNSZ
+	$LD     $tj,$tp,$BNSZ
+.align	4
+.Linner:
+	$ADDU   $aj,$ap,$j
+	$ADDU   $nj,$np,$j
+	$LD     $aj,$aj,0
+	$LD     $nj,$nj,0
+
+	$ADDU	$lo0,$alo,$hi0
+	$ADDU	$lo1,$nlo,$hi1
+	sltu	$t8,$lo0,$hi0
+	sltu	$t0,$lo1,$hi1
+	$ADDU	$hi0,$ahi,$t8
+	$ADDU	$hi1,$nhi,$t0
+	$MULD	$alo,$aj,$bi
+	$MULHD	$ahi,$aj,$bi
+
+	$ADDU	$lo0,$lo0,$tj
+	addi.d	$j,$j,$BNSZ
+	sltu	$t8,$lo0,$tj
+	$ADDU	$lo1,$lo1,$lo0
+	$ADDU	$hi0,$hi0,$t8
+	sltu	$t0,$lo1,$lo0
+	$LD	$tj,$tp,2*$BNSZ
+	$ADDU	$hi1,$hi1,$t0
+	sltu	$t8,$j,$num
+	$MULD	$nlo,$nj,$m1
+	$MULHD	$nhi,$nj,$m1
+	$ST     $lo1,$tp,0
+	$PTR_ADD $tp,$tp,$BNSZ
+	bnez	$t8,.Linner
+
+	$ADDU	$lo0,$alo,$hi0
+	sltu	$t8,$lo0,$hi0
+	$ADDU	$hi0,$ahi,$t8
+	$ADDU	$lo0,$lo0,$tj
+	sltu	$t0,$lo0,$tj
+	$ADDU	$hi0,$hi0,$t0
+
+	$LD     $tj,$tp,2*$BNSZ
+	$ADDU	$lo1,$nlo,$hi1
+	sltu	$t8,$lo1,$hi1
+	$ADDU	$hi1,$nhi,$t8
+	$ADDU	$lo1,$lo1,$lo0
+	sltu	$t0,$lo1,$lo0
+	$ADDU	$hi1,$hi1,$t0
+	$ST     $lo1,$tp,0
+
+	$ADDU	$lo1,$hi1,$hi0
+	sltu	$hi1,$lo1,$hi0
+	$ADDU	$lo1,$lo1,$tj
+	sltu	$t8,$lo1,$tj
+	$ADDU	$hi1,$hi1,$t8
+	$ST     $lo1,$tp,$BNSZ
+	$ST	$hi1,$tp,2*$BNSZ
+
+	$PTR_ADD $i,$i,$BNSZ
+	sltu	$t0,$i,$num
+	bnez	$t0,.Louter
```

Comment:
MIPS doesn't have fused compare-and-branch instructions like `bgtu/bleu/bgeu/bltu` so the MIPS code had to `sltu + bnez`. Please simplify all similar occurrences.